src/main/java/core/basesyntax/service/strategy/FruitTransactionStrategyImpl.java

```
@@ -0,0 +1,17 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.model.Operation;
+import java.util.Map;
+
+public class FruitTransactionStrategyImpl implements FruitTransactionStrategy {
+    private final Map<Operation, FruitTransactionHandler> map;
+
+    public FruitTransactionStrategyImpl(Map<Operation, FruitTransactionHandler> map) {
+        this.map = map;

```

Constructor does not validate the injected `map`. The checklist and description require null-parameter checks and failing fast with informative exceptions. Add a null check and throw a descriptive RuntimeException if `map` is null (for example: `if (map == null) { throw new RuntimeException("Operation handlers map must not be null"); }`). This prevents NPEs later and follows the dependency-check guidance.